### PR TITLE
Issue143

### PR DESCRIPTION
**Pull request by jnicholCU**
Newsletter page type addition. Does not have functionality to make the inlined css newsletter with php yet.
Has related pull requests in the paragraphs and page_types repos

Linked to #143

### CODE REVIEW

**Comment by jnicholCU** [comment]
Related Pull Requests:

https://github.com/CuBoulder/ucb_custom_page_types/pull/26
https://github.com/CuBoulder/ucb_custom_paragraphs/pull/23

File: templates/content/node--ucb-newsletter.html.twig [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+%}
+
+{#Dummy variable to ensure that all content tags are set for caching purposes#}
+{% set content_render = content|render %}
+
+<article{{attributes.addClass(classes)}}>
+	<div class="container ucb-newsletter">
+		{# Newsletter Title #}
+		<h1 class="ucb-newsletter-title" itemprop="name">{{ content.field_newsletter_type }}</h1>
+
```
Comment (patrickbrown-io) [inline templates/content/node--ucb-newsletter.html.twig:28]: Needs the following : Date, View on Website link 
#143 see render order

Comment (jnicholCU) [reply templates/content/node--ucb-newsletter.html.twig:28]: Good catch! Definitely needs the date.
The "View on Website" link will be part of the rendering for the email version of the page.

Comment (jnicholCU) [reply templates/content/node--ucb-newsletter.html.twig:28]: Oh actually, just check the website render on the webexpress instructions.
Doesn't look like the date gets on that page either. Just for the emailed version.

https://websupport.colorado.edu/article/261-newsletter

File: templates/content/node--ucb-newsletter.html.twig [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+				{% if content.field_newsletter_block_title_two|render %}
+					<h3>{{ content.field_newsletter_block_title_two }}</h3>
+				{% endif %}
+
+				{% if content.field_newsletter_block_text_two|render %}
+					{{ content.field_newsletter_block_text_two }}
+				{% endif %}
+			</div>
+		</div>
+	</div>
```
Comment (patrickbrown-io) [inline templates/content/node--ucb-newsletter.html.twig:67]: Needs a footer that is unique to this newsletter type

**Review by patrickbrown-io — COMMENTED** [review]
Functionality from admin side works following Kevin's specifications outlined on #143 

Needs:
- Date and Link to render after newsletter name
- Footer

**Comment by patrickbrown-io** [comment]
One last one - the teaser thumbnail image is on the right on current teaser content, Kevin's mockup shows teaser-style content with image on left

**Comment by jnicholCU** [comment]
> One last one - the teaser thumbnail image is on the right on current teaser content, Kevin's mockup shows teaser-style content with image on left

I just pinged Kevin about this one. His mockup seems different from what the webexpress site and options state in the creation instructions so Im going to get some more clarity 

The creation instructions state 
> Display: choose how articles in this section will display.
> Teaser: a 1/5 display with an article summary and a small thumbnail image aligned to the right.

And on the example addons page it has them displayed that way as well
https://www.colorado.edu/demos/web-express/newsletter/add-bundles/newsletter-example-one

Need to see if the teasers switch for the email view.

**Comment by jnicholCU** [comment]
I also just noticed that the Promos section for the Newsletter just don't exist in this build for some reason? Looks like all the files are missing so I must have failed to commit those up.

**Comment by patrickbrown-io** [comment]
I can test again once the promos are in

**Review by patrickbrown-io — COMMENTED** [review]
Awesome -- promos work on fresh build and appear in the correct order in the newsletter, as specified! I agree with you that it looks a bit cluttered depending on how much content is put in the sections, so that extra css would help give everything a little breathing room.

 #143 
"Each newsletter has 2 promo areas which include an image and a link." =>  we might need a promo link / button instead of promo text? Or in addition to the promo text? I'm not sure which is the best course, no example of promos on [web express demos](https://www.colorado.edu/demos/web-express/newsletter/add-bundles/newsletter-example-one)

**Comment by jnicholCU** [comment]
Wow yeah I don't know how I managed to build that wrong twice with text instead of a link. I believe the link will go on the promo so it's like an ad rather than a button. I'll get that changed up now so it's good for testing by the time you guys are up.

**Comment by jnicholCU** [comment]
Updated with the promo links added in. Should be working fine.

**Review by patrickbrown-io — COMMENTED** [review]
Promo images are clickable and in the correct place in the newsletter. CSS addition looks great. One thing, the teaser-style sections specifically on mobile screen size, might need to stack on mobile - it just gets VERY small fixed in a row. Other than that, looks great!

**Comment by jnicholCU** [comment]
Added the change for the teaser layout so that those stack better on mobile
changed from `"col-5" & "col-1"` to `"col-lg-5 col-8" & "col-lg-1 col-4"`

**Review by patrickbrown-io — APPROVED** [review]
Nice! The mobile responsive teasers look great, approve!

**Comment by jacob-korf** [comment]
Works great! Will be merged in

**Pull request merged**